### PR TITLE
chore: extract WireLogger into Swift package - WPB-11881

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11881" title="WPB-11881" target="_blank">TaskWPB-11881</a>  move Wirelogger to WireFoundation
  </td></table>
  <br />
 

### Issue

In order to be able to use WireLogger from Swift packages, this PR moves all WireLogger related code (except WireLoggerObjC) from WireSystem to its own target WireLogging.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: WireFoundation/Sources/WireFoundation/Logging/SystemLogger.swift [hunk 1]
```diff
@@ -66,16 +66,16 @@ struct SystemLogger: LoggerProtocol {
...
 
     func addTag(_ key: LogAttributesKey, value: String?) {
-       // do nothing, as it's only available on datadog
+        // do nothing, as it's only available on datadog
     }
 
-    private func log(_ message: LogConvertible, attributes: [LogAttributes], osLogType: OSLogType) {
+    private func log(_ message: any LogConvertible, attributes: [LogAttributes], osLogType: OSLogType) {
         var mergedAttributes: LogAttributes = [:]
-        attributes.forEach {
```
Comment (netbe) [inline WireFoundation/Sources/WireFoundation/Logging/SystemLogger.swift]: why the change?

File: WireUI/Package.resolved [hunk 2]
```diff
@@ -1,19 +1,19 @@
 {
-  "originHash" : "4f1f0d62407d5ee7a355f7111a27d6e49ae5b050e52bf6cdbca00fd5d9be585e",
+  "originHash" : "ac941621e4353f6ea1de2917885b39a50d5770cb8bc4e76af0db2004c38b1d41",
```
Comment (netbe) [inline WireUI/Package.resolved]: why is it changed?

File: wire-ios-system/Source/Logging/WireLogger.swift [hunk 3]
```diff
@@ -16,84 +16,6 @@
...
-
-    public static var logFiles: [URL] {
-        provider.logFiles
-    }
-
-    public static func addLogger(_ logger: LoggerProtocol) {
-        provider.addLogger(logger)
-    }
-}
+public typealias WireLogger = WireFoundation.WireLogger
```
Comment (netbe) [inline wire-ios-system/Source/Logging/WireLogger.swift]: that might be confusing, though a good idea to have a default logger here. Does it have to be public?
maybe `WireLoggerDefault`?

**Review by netbe — COMMENTED** [review]
added some comments, but otherwise good we should move forward on this, as it's needed on multiple packages

File: WireUI/Package.resolved (hunk 2 above)
Comment (caldrian) [reply WireUI/Package.resolved]: Closing this outdated comment.

File: WireFoundation/Sources/WireFoundation/Logging/SystemLogger.swift (hunk 1 above)
Comment (caldrian) [reply WireFoundation/Sources/WireFoundation/Logging/SystemLogger.swift]: Closing this outdated comment.

File: WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -3,171 +3,25 @@
...
-		594AF2062CF7358F00478C64 /* AllTests.xctestplan in Resources */ = {isa = PBXBuildFile; fileRef = 594AF2042CF7358F00478C64 /* AllTests.xctestplan */; };
+		59537CDD2CFF90D100920B59 /* WireLogging in Frameworks */ = {isa = PBXBuildFile; productRef = 59537CDC2CFF90D100920B59 /* WireLogging */; };
 		598D042D2C89C63100B64D71 /* WireFoundation in Frameworks */ = {isa = PBXBuildFile; productRef = 598D042C2C89C63100B64D71 /* WireFoundation */; };
 		59909A5E2C5BBEA8009C41DE /* WireAPI in Frameworks */ = {isa = PBXBuildFile; productRef = 59909A5D2C5BBEA8009C41DE /* WireAPI */; };
 		59909A692C5BC001009C41DE /* WireAPI in Frameworks */ = {isa = PBXBuildFile; productRef = 59909A682C5BC001009C41DE /* WireAPI */; };
 		59909A6D2C5BC08B009C41DE /* WireAPISupport in Frameworks */ = {isa = PBXBuildFile; productRef = 59909A6C2C5BC08B009C41DE /* WireAPISupport */; };
-		C91F111C2C9C4B2B00BA5BE2 /* ConnectionsRepository.swift in Sources */ = {isa = PBXBuildFile; fileRef = C91F111A2C9C4B2B00BA5BE2 /* ConnectionsRepository.swift */; };
```
Comment (caldrian) [inline WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj:24]: Converted groups to folders. A few missing files have been added this way as well.

File: WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -3,171 +3,25 @@
...
-		C9D579442CF61F6D00012A0E /* ConversationMessageTimerUpdateEventProcessorTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = C9D579432CF61F6D00012A0E /* ConversationMessageTimerUpdateEventProcessorTests.swift */; };
-		C9D574582CEB9FB400012A0E /* ConversationRenameEventProcessorTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = C9D574572CEB9FB400012A0E /* ConversationRenameEventProcessorTests.swift */; };
-		C9D57A8A2CF732A500012A0E /* MessageLocalStoreTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = C98433F42CCBC290009723D4 /* MessageLocalStoreTests.swift */; };
-		C9D57A932CF7556800012A0E /* ConversationRepositoryTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = C9D57A922CF7556800012A0E /* ConversationRepositoryTests.swift */; };
-		C9D57B232CF75B3700012A0E /* FeatureConfigRepositoryTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = C9E8A3BF2C761EDD0093DD5C /* FeatureConfigRepositoryTests.swift */; };
+		C9D57A8A2CF732A500012A0E /* (null) in Sources */ = {isa = PBXBuildFile; };
```
Comment (netbe) [inline WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj:22]: I guess this can go too

File: WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -1151,171 +348,22 @@
...
-				C97C01AC2CB92D47000683C5 /* UserLegalholdEnableEventProcessorTests.swift in Sources */,
-				C96B75652CDCC85B003A85EB /* TeamLocalStoreTests.swift in Sources */,
-				C96B75702CDD0000003A85EB /* MockFeatureConfigLocalStoreProtocol.swift in Sources */,
-				C97C015A2CB40010000683C5 /* FederationDeleteEventProcessorTests.swift in Sources */,
-				C9C8FDD42C9DBE0E00702B91 /* UserLegalholdRequestEventProcessorTests.swift in Sources */,
-				C96B755D2CDBB176003A85EB /* ConversationLocalStoreTests.swift in Sources */,
-				C96B75772CDD154C003A85EB /* UpdateEventsLocalStoreTests.swift in Sources */,
-				017F679C2C20801800B6E02D /* TeamRepositoryTests.swift in Sources */,
-				C9E8A3B72C749F2A0093DD5C /* ConversationLabelsRepositoryTests.swift in Sources */,
+				C9D57A8A2CF732A500012A0E /* (null) in Sources */,
```
Comment (netbe) [inline WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj:366]: removed ?

File: WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj (hunk 5 above)
Comment (caldrian) [reply WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj:22]: done

**Review by netbe — COMMENTED** [review]
We need to update framework.rb so for example WireSystem is tested when WireLogger is touched.

For now I would consider WireFoundation to be WireLogger, unless @caldrian you want to go the extra step of introducing WireLogger as it's on framework

**Comment by caldrian** [comment]
> unless caldrian you want to go the extra step of introducing WireLogger as it's on framework

made WireLogging a separate Swift package @netbe

File: WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj (hunk 5 above)
Comment (caldrian) [reply WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj:22]: I had to revert it, it broke the project.

File: WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj (hunk 6 above)
Comment (caldrian) [reply WireDomain/Project/WireDomain Project.xcodeproj/project.pbxproj:366]: I had to fix the project, than again converted the groups to folders and Xcode seems to create this because it's needed. So I'll keep it.

File: WireAnalytics/Sources/WireAnalytics/Service/AnalyticsService.swift [hunk 7]
```diff
@@ -44,7 +44,7 @@ public final class AnalyticsService: AnalyticsServiceProtocol {
 
     public convenience init(
         config: Config?,
-        logger: @escaping (String) -> Void
+        logger: @escaping (String) -> Void // TODO: [WPB-11881] import WireLogging and remove this workaround
```
Comment (caldrian) [inline WireAnalytics/Sources/WireAnalytics/Service/AnalyticsService.swift:47]: PR: https://github.com/wireapp/wire-ios/pull/2253

File: wire-ios-system/Source/Logging/WireLogger.swift (hunk 3 above)
Comment (caldrian) [reply wire-ios-system/Source/Logging/WireLogger.swift]: This code has been removed, so the comment is outdated.

File: fastlane/framework.rb [hunk 8]
```diff
@@ -20,11 +28,6 @@ def self.all
             "wire-ios-transport",
             "wire-ios-utilities",
             "wire-ios-ziphy",
-            "WireAPI",
-            "WireAnalytics",
-            "WireDomain",
-            "WireUI",
-            "WireFoundation",
```
Comment (caldrian) [inline fastlane/framework.rb:27]: sorted

File: WireLogging/Sources/WireLoggingSupport/WireDomainPackage.swift
Comment (netbe) [inline WireLogging/Sources/WireLoggingSupport/WireDomainPackage.swift:1]: rename to WireLoggingPackage.swift

File: fastlane/framework.rb [hunk 9]
```diff
@@ -3,6 +3,14 @@ class Framework
...
     def self.all 
         all_folders = [
+            "WireAPI",
+            "WireAPI",
+            "WireAnalytics",
+            "WireDomain",
+            "WireFoundation",
+            "WireLogging",
+            "WireUI",
+            "WireUI",
```
Comment (netbe) [inline fastlane/framework.rb]:
```suggestion
            "WireAPI",
            "WireAnalytics",
            "WireDomain",
            "WireFoundation",
            "WireLogging",
            "WireUI",
```

File: fastlane/framework.rb [hunk 10]
```diff
@@ -36,43 +39,58 @@ def self.all
...
         frameworks["wire-ios-utilities"].add_dependency(frameworks["wire-ios-system"])
         frameworks["wire-ios-utilities"].add_dependency(frameworks["WireFoundation"])
-        
+        frameworks["wire-ios-utilities"].add_dependency(frameworks["WireLogging"])
+
         frameworks["wire-ios-testing"].add_dependency(frameworks["wire-ios-system"])
-        
+
+        frameworks["wire-ios-system"].add_dependency(frameworks["WireLogging"])
+
```
Comment (netbe) [inline fastlane/framework.rb:86]: need to update also the to_scheme to return WireLogging

**Review by netbe — COMMENTED** [review]
I thought we could keep WireLogging inside WireFoundation but this works too for now.

a couple small things remain:
1. the detection for .github/workflows/test_pr_changes.yml
2. to_scheme in framework.rb
3. the AllTests testplan for WireLogging

File: fastlane/framework.rb (hunk 10 above)
Comment (caldrian) [reply fastlane/framework.rb:86]: done

File: WireLogging/Sources/WireLoggingSupport/WireDomainPackage.swift
Comment (caldrian) [reply WireLogging/Sources/WireLoggingSupport/WireDomainPackage.swift:1]: good catch, thanks

**Review by netbe — APPROVED** [review]
🚢

**Review by KaterinaWire — APPROVED** [review]
